src/reference/asciidoc/xmpp.adoc

```diff
@@ -80,6 +80,42 @@ When started it will register a `PacketListener` that will listen for incoming X
 It forwards any received messages to the underlying adapter which will convert them to Spring Integration Messages and send them to the specified `channel`.
 It will unregister the `PacketListener` when it is stopped.
 
+Starting with _version 4.3_ the `ChatMessageListeningEndpoint` (and its `<int-xmpp:inbound-channel-adapter>`)
+supports a `org.jivesoftware.smack.filter.StanzaFilter` injection to be registered on the provided `XMPPConnection`
+together with an internal `StanzaListener` implementation.
+See their https://www.igniterealtime.org/builds/smack/docs/latest/javadoc/org/jivesoftware/smack/XMPPConnection.html#addAsyncStanzaListener%28org.jivesoftware.smack.StanzaListener,%20org.jivesoftware.smack.filter.StanzaFilter%29[JavaDocs] for more information.
+
+Also with the _version 4.3_ the `payload-expression` has been introduced for the `ChatMessageListeningEndpoint`.
+The incoming `org.jivesoftware.smack.packet.Message` represents a root object of evaluation context.
+This option is useful in case of <<xmpp-extensions>>.
+For example, for the GCM protocol we can extract the body using expression:
+
+[source,xml]
+----
+payload-expression="getExtension('google:mobile:data').json"
```

Comment:
With the Spring MVC you have the particular Controller's method with specific (domain) parameter. So, MVC infrastructure knows how to convert an incoming JSON to the tarter object.

With Spring Integration we deal with *messages* and any that `target type` is up to the target application. It is fully outside of the Inbound Channel Adapter responsibility. 
The EIP solution based on the messaging pursue the **loosely coupled** principle as a main feature in between components. 

That's why I still insist that the XMPP Adapter should not know anything about your domain and just produces as convenient message as possible.
See my comment about the `<int:json-to-object-transformer>`, who is really responsible to produce a domain object for the incoming JSON `payload`.

From other hand, don't forget, that we aim to support as much generic XMPP as possible, that's why another my sample in the documentation demonstrates `XHTML`, where the target `payload` is fully different from the JSON and produces an XML-like structure. For this purpose we can use an `<int-xml:unmarshalling-transformer>`.

Please, elaborate more about your last sentence. What you would like to see in the Doc on the matter?
If you are familiar with the `GcmExtension` and know how to extract it from the XMPP `Message`, that would not be so difficult to determine that the `json` in the SpEL is just a getter from that `GcmExtension` object. And that is fully clear that you have a `String` `payload` as an output of the adapter. 

Actually that is true for any Spring Integration Inbound Channel Adapter (or even any producing `MessageHandler`), that you should keep in mind what type is in the message to determine that further logic in your flow. That is some minor inconvenient side-effect of the  loosely coupled principle, when we don't know the type during the compilation time and always should track that manually to make the proper routing logic. The Messaging framework, like Spring Integration, doesn't care about `payload`, because it fully outside of Messaging architecture and really should not be included there.